fix/improve backward/forward compatibility for VM Backup and Restore (#731)
harvester/docs: versioned_docs/version-v1.3/advanced/csidriver.md

1. Find and select **csi-driver-config**, and then select **⋮** > **Edit Setting** to access the configuration options.
1. Set the **Provisioner** to the third-party CSI driver in the settings.
1. Next, Configure the **Volume Snapshot Class Name**. This setting points to the name of the `VolumeSnapshotClass` used for creating volume snapshots or VM snapshots.

![csi-driver-config-external](/img/v1.2/advanced/csi-driver-config-external.png)

:::caution

> - Staring from **v1.4.2**, Harvester will block setting **Backup Volume Snapshot Class Name** for third-party storage, becuase except for **Longhorn V1 Data Engine**, **Backup** is not included in the CSI standard. There is no general way to have a remote backup for all kinds of storage providers.

Reviewer: ```suggestion
- Once upgraded to **v1.4.2 or higher**, Harvester will block setting **Backup Volume Snapshot Class Name** for third-party storage, because except for **Longhorn V1 Data Engine**, **Backup** is not included in the CSI standard. There is no general way to have a remote backup for all kinds of storage providers.
```

I think it will be clearer if we articulate this from an upgrade perspective. Same for `settings.md` and `csidriver.md`.

> - If the **VM Backup** with third-party storage was already created mistakenly before **v1.4.2**, Harvester will refuse to restore it, since the data never has a remote copy.

Reviewer: ```suggestion
- Harvester will be unable to restore all the existing VM backups with third-part storage, since the data never has a remote copy.
```
